style_guides/accessibility_guide.md

```diff
@@ -0,0 +1,230 @@
+otherwise. You won't need this often.
+
+*Why should you add an `id`?*
+
+Due to some bugs in some common screen readers, you will always need to add an `id`
+to an element with `tabindex`, since they wouldn't pick up the `tabindex` correctly
+otherwise.
+
+### Tooltips
+
+**TL;DR** *Add `role="tooltip"` and `aria-describedby` to elements for accessible tooltips.*
+
+Elements which act as tooltips should have the `role="tooltip"` attribute and an ID to which the
+described element can point to with the `aria-describedby` attribute. For example:
+
+```html
+<div
+  class="kuiTooltip"
+  role="tooltip"
+  id="visualizationsTooltip"
+>
+  Visualizations help you make sense of your data.
+</div>
+
+<button aria-describedby="visualizationsTooltip">
+ Visualizations
+</button>
+```
+
+### Don't create keyboard traps
+
+**TL;DR** *If you can't leave an element with <kbd>Tab</kbd> again, it needs a special interaction model.*
+
+If an interactive element consumes the <kbd>Tab</kbd> key (e.g. a code editor to
+create an actual tabular indentation) it will prevent a keyboard user to leave
+that element again. Also see [WCAG 2.1.2](https://www.w3.org/TR/WCAG20/#keyboard-operation-trapping).
+
+Those kind of elements, require a special interaction model. A [code editor](https://github.com/elastic/kibana/pull/13339)
+could require an <kbd>Enter</kbd> keypress before starting editing mode, and
+could leave that mode on <kbd>Escape</kbd> again.
+
+Unfortunately there is no universal solution for this problem, so be aware when creating
+such elements, that would consume tabbing, to think about an accessible interaction
+model.
+
+*Hint:* If you create that kind of interactive elements `role="application"` might
+be a good `role` (also see below) for that element. It is meant for elements providing
+their own interaction schemes.
+
+## Roles
+
+Each DOM element has an implicit role in the accessibility tree (that assistive technologies
+use). The mapping of elements to default roles can be found in the
+[Accessibility API Mappings](https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings).
+You can overwrite this role via the `role` attribute on an element, and the
+assistive technology will now behave the same like any other element with that role
+(e.g. behave like it is a button when it has `role="button"`).
+
+### Landmark roles
+
+Some roles can be used to declare so called landmarks. These landmarks tag important
+parts of a web page. Screen readers offer a quick way to jump to these
+parts of the page (*landmark navigation*).
+
+#### role=main
+
+The `main` role (or equivalent using the `<main>` tag) declares the main part
+of a page. This can be used in the landmark navigation to quickly jump to the
+actual main area of the page (and skip all headers, navigations, etc.).
+
+#### `<section>`
+
+The `<section>` element, can be used to mark a region on the page, so that it
+appears in the landmark navigation. The section element therefor needs to have
```

Comment:
Just wanted to sound [archaic](https://en.oxforddictionaries.com/definition/therefor) - no kidding, will fix it.